store/storethehash/option.go

```diff
@@ -0,0 +1,49 @@
+package storethehash
+
+import (
+	"time"
+
+	sthtypes "github.com/ipld/go-storethehash/store/types"
+)
+
+// TODO: Benchmark and fine-tune for better performance.
+const (
+	defaultBurstRate     = 4 * 1024 * 1024
+	defaultIndexSizeBits = uint8(24)
+	defaultSyncInterval  = time.Second
+)
+
+// config contains all options for configuring storethehash valuestore.
+type config struct {
+	burstRate     sthtypes.Work
+	indexSizeBits uint8
+	syncInterval  time.Duration
+}
+
+type Option func(*config)
+
+// apply applies the given options to this config.
+func (c *config) apply(opts []Option) {
+	for _, opt := range opts {
+		opt(c)
+	}
+}
+
+func IndexBitSize(indexBitSize uint8) Option {
```

Comment:
do you want to hook up a way for these to get set as part of this?

Reply:
The defaults are applied (line 62 in `storethehash.go` in the PR), and these can be set in the indexer when it creates the storethehash valuestore, that gets passed to the indexer-core constructor.